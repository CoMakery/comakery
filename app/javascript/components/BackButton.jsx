import React from 'react'
import Icon from './styleguide/Icon'
import styled from 'styled-components'

const Wrapper = styled.div`
  position: absolute;
  top: 20px;
  left: 20px;
  cursor: pointer;

  @media (max-width: 1024px) {
    display: none;
  }
`

class BackButton extends React.Component {
  goBack() {
    typeof window === 'undefined' ? null : window.location = document.referrer
  }

  render() {
    return (
      <React.Fragment>
        <Wrapper>
          <Icon name="iconCloseCopy.svg" onClick={this.goBack} />
        </Wrapper>
      </React.Fragment>
    )
  }
}

export default BackButton
