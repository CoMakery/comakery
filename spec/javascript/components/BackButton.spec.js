import React from 'react'
import { mount } from 'enzyme'
import BackButton from 'components/BackButton'

describe('BackButton', () => {
  it('renders component correctly', () => {
    const wrapper = mount(<BackButton />)

    expect(wrapper).toMatchSnapshot()
  })
})
