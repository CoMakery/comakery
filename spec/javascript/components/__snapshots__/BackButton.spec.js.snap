// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton renders component correctly 1`] = `
<BackButton>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "fHEVVg",
            "rules": Array [
              "
  position: absolute;
  top: 20px;
  left: 20px;
  cursor: pointer;

  @media (max-width: 1024px) {
    display: none;
  }
",
            ],
          },
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa fHEVVg"
      >
        <Icon
          className=""
          name="iconCloseCopy.svg"
          onClick={[Function]}
        >
          <img
            className="icon icon__iconCloseCopy-svg"
            onClick={[Function]}
            src="test-file-stub"
          />
        </Icon>
      </div>
    </StyledComponent>
  </styled.div>
</BackButton>
`;
